private/buf/cmd/buf/command/generate/generate.go

```diff
@@ -350,46 +334,152 @@ func run(
 	if err != nil {
 		return err
 	}
-	imageConfigs, fileAnnotations, err := imageConfigReader.GetImageConfigs(
-		ctx,
-		container,
-		ref,
-		flags.Config,
-		flags.Paths,        // we filter on files
-		flags.ExcludePaths, // we exclude these paths
-		false,              // input files must exist
-		false,              // we must include source info for generation
-	)
-	if err != nil {
-		return err
+
+	var includedTypesFromCLI []string
+	if len(flags.Types) > 0 || len(flags.TypesDeprecated) > 0 {
+		// command-line flags take precedence
+		includedTypesFromCLI = append(flags.Types, flags.TypesDeprecated...)
 	}
-	if len(fileAnnotations) > 0 {
-		if err := bufanalysis.PrintFileAnnotations(container.Stderr(), fileAnnotations, flags.ErrorFormat); err != nil {
+
+	var (
+		inputImages   []bufimage.Image
+		imageModifier bufimagemodify.Modifier
+		plugins       []bufgen.PluginConfig
+	)
+	switch configVersion {
+	case bufgen.V2Version:
```

Comment:
Does v2 actually work after this PR? If not, we shouldn't merge in a state to main that will accept it, we should still error until all the work is complete

Reply:
This actually merges into `bufgenv2`, and so I think it's ok to do this.
I can also merge into main, but I think it's easy to experiment with (and catch bugs) in such a state.